src/unix/os390.c

```diff
@@ -665,11 +667,120 @@ int uv__io_check_fd(uv_loop_t* loop, int fd) {
   return 0;
 }
 
+
+void uv__fs_event_close(uv_fs_event_t* handle) {
+  uv_fs_event_stop(handle);
+}
+
+
+int uv_fs_event_init(uv_loop_t* loop, uv_fs_event_t* handle) {
+  // Register handle
+  uv__handle_init(loop, (uv_handle_t*)handle, UV_FS_EVENT);
+  return 0;
+}
+
+
+int uv_fs_event_start(uv_fs_event_t* handle, uv_fs_event_cb cb,
+                      const char* filename, unsigned int flags) {
+  uv__os390_epoll* ep;
+  _RFIS reg_struct;
+  int saved_errno;
+  int rc;
+
+  ep = handle->loop->ep;
+  assert(ep->msg_queue != -1);
+
+  reg_struct.__rfis_cmd  = _RFIS_REG;    
+  reg_struct.__rfis_qid  = ep->msg_queue;     
+  reg_struct.__rfis_type = 1;            
+  memcpy(reg_struct.__rfis_utok, &handle, sizeof(handle));
+
+  rc = __w_pioctl(filename, _IOCC_REGFILEINT, sizeof(reg_struct),
+                  &reg_struct);
+  if (rc != 0)
+    return -errno;
+
+  /* Start handle. */
+  uv__handle_start(handle);
+  memcpy(handle->rfis_rftok, reg_struct.__rfis_rftok, sizeof(handle->rfis_rftok));
+  handle->path = uv__strdup(filename);
+  if (handle->path == NULL) {
+    saved_errno = errno;
+    reg_struct.__rfis_cmd  = _RFIS_UNREG;    
+    reg_struct.__rfis_qid  = ep->msg_queue;     
+    reg_struct.__rfis_type = 1;            
+    __w_pioctl(NULL, _IOCC_REGFILEINT, sizeof(reg_struct),
+               &reg_struct);
+    return -saved_errno;
```

Comment:
I will perform the strdup before the initial __w_pioctl call.
Then when strdup fails, I don't need to unregister by calling __w_pioctl again.